sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/TracerProvider.java

```diff
@@ -0,0 +1,206 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.implementation;
+
+import com.azure.core.util.Context;
+import com.azure.core.util.FluxUtil;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.util.tracing.Tracer;
+import com.azure.cosmos.CosmosClientException;
+import com.azure.cosmos.models.CosmosAsyncItemResponse;
+import com.azure.cosmos.models.CosmosResponse;
+import reactor.core.publisher.Mono;
+import reactor.core.publisher.Signal;
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Function;
+
+import static com.azure.core.util.tracing.Tracer.AZ_TRACING_NAMESPACE_KEY;
+
+public class TracerProvider {
+    private final ClientLogger logger = new ClientLogger(TracerProvider.class);
+    private final List<Tracer> tracers = new ArrayList<>();
+    public final static String DB_TYPE_VALUE = "cosmosdb";
+    public final static String DB_TYPE = "db.type";
+    public final static String DB_INSTANCE = "db.instance";
+    public final static String DB_URL = "db.url";
+    public static final String DB_STATEMENT = "db.statement";
+    public static final String ERROR_MSG = "error.msg";
+    public static final String ERROR_TYPE = "error.type";
+    public static final String ERROR_STACK = "error.stack";
+    public static final String COSMOS_CALL_DEPTH = "cosmosCallDepth";
+    public static final String MICROSOFT_DOCOMENTDB = "Microsoft.DocumentDB";
```

Comment:
Does documentDB refer to us? if so, that name is deprecated.

Reply:
The resource provider name is being provided from this [list of Azure services](https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/azure-services-resource-providers).
Feature request to include this on all Track 2 SDK's - https://github.com/Azure/azure-sdk-for-java/issues/7049

Reply:
@samvaity it seems as per the link you shared the service provider name of our service is not updated from Microsoft.DocumentDB to Microsoft.Cosmos.

My question to you is if for tracer work, this is merge implementation details or whether in the application insight the customers will see the term "DocumentDB" ?

Reply:
FYI @kirankumarkolli

Reply:
The customers would see the term "DocumentDB".
I think the ask was from the Azure Monitor team to be able to group spans/trace information for a particular Azure service and wanted to follow consistent naming across different components (i. Azure Monitor). 

>Azure Monitor exporter will use this information to navigate to the corresponding Azure resource from AzMon UX and show proper icons. #7049

I agree we should not follow deprecated naming but then is this listed incorrectly on the [list](https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/azure-services-resource-providers) too?
@trask Do you think if Cosmos should still be referred to as `Microsoft.DocumentDB`?

Reply:
Our SDK names already deviate from the RP name. 
Just for context: "Azure Cosmos DB" is the new branding (and RP will rebrand eventually). 

For all Track2 libraries: How about using the package name 'Azure.Cosmos" in our specific case?

/cc: @JonathanGiles

Reply:
Discussed with Sameeksha , once the documentation [here ](https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/azure-services-resource-providers) updates with azure.cosmosdb, we will change in the code too. Other azure services are also mentioned as Microsoft currently.